server/src/main/java/org/elasticsearch/cluster/NodeExecutionLoad.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.cluster;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Record of a node's thread pool usage stats. Maps thread pool stats by thread pool name.
+ *
+ * @param nodeId The node ID.
+ * @param threadPoolUsageStatsMap A map of thread pool name ({@link org.elasticsearch.threadpool.ThreadPool.Names}) to the thread pool's
+ *                                usage stats ({@link ThreadPoolUsageStats}).
+ */
+public record NodeExecutionLoad(String nodeId, Map<String, ThreadPoolUsageStats> threadPoolUsageStatsMap) implements Writeable {
```

Comment:
How about we use the phrase `ThreadPoolUsageStats`? I think we can also simplify the name of the map field.
```suggestion
public record NodeThreadPoolUsageStats(String nodeId, Map<String, ThreadPoolUsageStats> threads) implements Writeable {
```